db/rdbms/migration/0003_add_jobs_state_column.sql

```diff
@@ -0,0 +1,63 @@
+-- Copyright (c) Facebook, Inc. and its affiliates.
+--
+-- This source code is licensed under the MIT license found in the
+-- LICENSE file in the root directory of this source tree.
+
+-- +goose Up
+
+ALTER TABLE jobs ADD COLUMN state tinyint DEFAULT 0 AFTER extended_descriptor;
+CREATE INDEX job_state ON jobs (state, job_id);
+
+-- Populate the column.
+
+SET SQL_BIG_SELECTS=1;
```

Comment:
it may take a while, for sure, but it won't explode the database as such.
fwiw, i tested with a dump of current xdb.contest and it takes about 20 seconds.